src/main/java/io/smartcat/ranger/core/parser/DataGenerator.java

```diff
@@ -0,0 +1,108 @@
+package io.smartcat.ranger.core.parser;
+
+import io.smartcat.ranger.core.CompositeValue;
+import io.smartcat.ranger.core.PrimitiveValue;
+import io.smartcat.ranger.core.Value;
+import io.smartcat.ranger.core.ValueProxy;
+import org.parboiled.Parboiled;
+import org.parboiled.parserunners.ReportingParseRunner;
+import org.parboiled.support.ParsingResult;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Generates data specified by configuration.
+ */
+public class DataGenerator {
+
+    private CompositeValue outputValue;
+
+    private DataGenerator(Map<String, ValueProxy<?>> proxyValues, String outputValueName) {
+        Value<?> val = proxyValues.get(outputValueName);
+        if (val instanceof CompositeValue) {
+            this.outputValue = (CompositeValue) val;
+        } else {
+            Map<String, Value<?>> values = new HashMap<>();
+            values.put("output", val);
+            this.outputValue = new CompositeValue(values);
+        }
+    }
+
+    /**
+     * Returns next value.
+     *
+     * @return Next value.
+     */
+    public Map<String, Object> next() {
+        outputValue.reset();
+        return outputValue.get();
+    }
+
+    /**
+     * Builds {@link DataGenerator} based on specified <code>config</code> and <code>outputValueName</code>.
+     *
+     * @param config Data generation configuration.
+     * @param outputValueName Name of the output value.
+     * @return An instance of {@link DataGenerator}.
+     */
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    public static DataGenerator build(Map<String, Object> config, String outputValueName) {
+        Map<String, ValueProxy<?>> proxyValues = new HashMap<>();
+        ValueExpressionParser parser = Parboiled.createParser(ValueExpressionParser.class, proxyValues);
+        ReportingParseRunner<Value<?>> parseRunner = new ReportingParseRunner<>(parser.value());
+
+        for (Map.Entry<String, Object> entry : config.entrySet()) {
```

Comment:
Could this block be avoided by calling `parse(Object def, ReportingParseRunner<Value<?>> parseRunner, Map<String, ValueProxy<?>> proxyValues)`?